pkg/drivers/kic/kic.go

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kic
+
+import (
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/docker/machine/libmachine/drivers"
+	"github.com/docker/machine/libmachine/state"
+	"github.com/pkg/errors"
+	pkgdrivers "k8s.io/minikube/pkg/drivers"
+	"k8s.io/minikube/pkg/drivers/kic/node"
+	"k8s.io/minikube/pkg/drivers/kic/oci"
+	"k8s.io/minikube/pkg/minikube/command"
+)
+
+// https://minikube.sigs.k8s.io/docs/reference/drivers/kic/
+type Driver struct {
+	*drivers.BaseDriver
+	*pkgdrivers.CommonDriver
+	URL           string
+	exec          command.Runner
+	OciBinary     string
+	ImageSha      string
+	CPU           int
+	Memory        int
+	APIServerPort int32
+}
+
+// Config is configuration for the kic driver
+type Config struct {
+	MachineName   string
+	CPU           int
+	Memory        int
+	StorePath     string
+	OciBinary     string // oci tool to use (docker, podman,...)
+	ImageSha      string // image name with sha to use for the node
+	APIServerPort int32  // port to connect to forward from container to user's machine
+}
+
+// NewDriver returns a fully configured Kic driver
+func NewDriver(c Config) *Driver {
+	d := &Driver{
+		BaseDriver: &drivers.BaseDriver{
+			MachineName: c.MachineName,
+			StorePath:   c.StorePath,
+		},
+		exec:          command.NewKICRunner(c.MachineName, c.OciBinary),
+		OciBinary:     c.OciBinary,
+		ImageSha:      c.ImageSha,
+		CPU:           c.CPU,
+		Memory:        c.Memory,
+		APIServerPort: c.APIServerPort,
+	}
+	return d
+}
+
+// Create a host using the driver's config
+func (d *Driver) Create() error {
+	ks := &node.Spec{ // kic spec
+		Profile:           d.MachineName,
+		Name:              d.MachineName,
+		Image:             d.ImageSha,
+		CPUs:              strconv.Itoa(d.CPU),           //TODO: change kic to take int
+		Memory:            strconv.Itoa(d.Memory) + "mb", // TODO: change kic to take int
+		Role:              "control-plane",
+		ExtraMounts:       []oci.Mount{},
+		ExtraPortMappings: []oci.PortMapping{},
+		APIServerAddress:  "127.0.0.1", // medyagh:TODO make configurable
+		APIServerPort:     d.APIServerPort,
+		IPv6:              false, // MEDYA:TODO add proxy envs here
+	}
+
+	_, err := ks.Create(command.NewKICRunner(d.MachineName, d.OciBinary))
```

Comment:
Apparently this PR doesn't use the node returned by ks.Create. I'm not a big fan of unused functionality: What is the intent here?